File: rice_web/mock/product.js
export default [
  {
    url: '/product/all',
    type: 'get',
    response: _ => {
      var products = [
        {
          "label": "青岛啤酒",
          "status": "ON_SALE",
          "owned_number": 5020,
          "last_supple_date": "2019-05-04 12:00:00",
          "stardand_price": 20
        },
        {
          "label": "青岛啤酒3",
          "status": "ON_SALE",
          "owned_number": 50320,
          "last_supple_date": "2019-05-04 12:00:00",
          "stardand_price": 203
        },
      ]
      return {
          code: 20000,
          data: products
      }
    }
  },
  {
    url: '/product/add',
    type: 'POST',
    response: _ => {
      return {
        code: 20000,
        data: ''
      }
    }
  }
]

File: rice_server/rice_server/views/product.py
from django.http import JsonResponse, HttpResponse
from rice_model.models import Product

import json


def all(request):
    productsQS = Product.objects.all()

    products = products_to_json(productsQS)
    print(products)
    return JsonResponse({
        'code': 20000,
        'data': products
    })


def add(request):
    dataStr = request.GET.get('data')
    data = json.loads(dataStr)

    label = ''.join(data.get('label'))
    last_supple_date = ''.join(data.get('last_supple_date')[:10])
    status = data.get('status')
    owned_number =  int(''.join(data.get('owned_number')))

    product = Product(
        label=label,
        last_supple_date=last_supple_date,
        owned_number=owned_number,
        status=status
    )
    product.save()
    return JsonResponse({
        'code': 20000,
        'data': "success"
    })


def products_to_json(products):
    list = []
    for product in products:
        list.append({
            'label': product.label,
            'owned_number': product.owned_number,
            'last_supple_date': product.last_supple_date,
            'status': product.status
        })
    return list
